### PR TITLE
Major refactoring of functions.py, removed need to athenticate when using /report, updated lezioni's input to work as esami 

### PR DESCRIPTION
**Pull request by Guberlo**
First of all.. sorry for the gigantic PR but I came up with the idea to refactor functions.py just after implementing the last two features 😅

I've already tested the major part of the bot (excluding gitlab/drive related functions, since I can't access those ).
Those are the functions/files that need to be tested:
callback_handlers.py: callback()
request.py: request(), add_db()
gitlab.py: *
gdrive.py: *

### CODE REVIEW

File: module/help.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+from telegram import Update, InlineKeyboardMarkup, InlineKeyboardButton
+from telegram.ext import CallbackContext
+
+import random
+
+from module.shared import check_log, CUSicon
+
+def help_cmd():
```
Comment (TendTo) [inline module/help.py:8]: Maybe this long string could be put in a markdown file, and you could even keep the random emoji with a placeholder, if needed

File: module/lezioni.py [hunk 2]
```diff
@@ -1,60 +1,202 @@
...
-    return switcher.get(anno, "Anno non valido, utilizzare primo/secondo/terzo")
-
-def lezioni_output(item):
-    output = "*Nome:* " + item["nome"]
-    output += "\n*Aula:* " + str(item["aula"])
-    output += "\n*Anno:* " + str(item["anno"]) + "\n"
+    return switcher.get(day, "Giorno non valido")
 
+# eliminate
+def lezioni_button():
```
Comment (TendTo) [inline module/lezioni.py:31]: Is this still used somewhere or can just be removed?

File: module/lezioni.py [hunk 3]
```diff
@@ -1,60 +1,202 @@
...
+            ]
+        )
+
+    return message_text, InlineKeyboardMarkup(keyboard)
+
+def lezioni(update: Update, context: CallbackContext):
+    check_log(update, context, "lezioni")
+    if 'lezioni' in context.user_data: context.user_data['lezioni'].clear() #ripulisce il dict dell'user relativo al comando /esami da eventuali dati presenti
+    else: context.user_data['lezioni'] = {} #crea il dict che conterrà i dati del comando /esami all'interno della key ['esami'] di user data
+
```
Comment (TendTo) [inline module/lezioni.py:67]: Some comments still refer to the /esami command. You may want to change that, for better clarity

File: module/lezioni.py [hunk 4]
```diff
@@ -1,60 +1,202 @@
...
+    message_id = update.callback_query.message.message_id
+    lezioni_user_data = context.user_data['lezioni']
+    if "anno" in callbackData:
+        if callbackData[20:] not in lezioni_user_data.keys(): #se non era presente, setta la key di [1° anno|2° anno| 3° anno] a true... 
+            lezioni_user_data[callbackData[20:]] = True 
+        else:
+           del lezioni_user_data[callbackData[20:]] #... o elmina la key se era già presente
+    elif "giorno" in callbackData:
+        if 'giorno' + callbackData[22:] not in lezioni_user_data.keys(): #se non era presente, setta la key di sessione[prima|seconda|terza] a true... 
+            lezioni_user_data['giorno' + callbackData[22:]] = True
```
Comment (TendTo) [inline module/lezioni.py:97]: It's not really an issue, but wouldn't the key be "giorno1 giorno" this way?

**Review by TendTo — COMMENTED** [review]
I was hoping someone brave enough to refactor that huge function.py, so that is already great. Now, the pr is big, but i gave a quick overview  on the parts I'm more familiar with and that's what I found

File: main.py [hunk 5]
```diff
@@ -102,6 +120,10 @@ def main():
 	dp.add_handler(MessageHandler(Filters.regex(r"^(?!=<[/])[Ii]ns:\s+"), esami_input_insegnamento)) #regex accetta [/ins: nome] oppure [/Ins: nome], per agevolare chi usa il cellulare
 	dp.add_handler(CallbackQueryHandler(esami_handler, pattern='esami_button_*'))
 
+	# lezioni
+	dp.add_handler(CallbackQueryHandler(lezioni_handler, pattern='lezioni_button_*'))
```
Comment (Helias) [inline main.py:124]: @TendTo are you sure?

File: module/help.py (hunk 1 above)
Comment (Helias) [reply module/help.py:8]: agree!

Comment (Helias) [reply module/help.py:8]: But actually, we can't put in the markdown the randomization of the CUS icons xD

Comment (TendTo) [reply module/help.py:8]: Just put a placeholder in the markdown (something like %placeholder%) and do a .replace("%placeholder%, icon)

File: module/lezioni.py (hunk 2 above)
Comment (Guberlo) [reply module/lezioni.py:31]: Nope, I completely forgot to delete this 😅
I don't even remember why I have commented but not actualy deleted this...

File: module/lezioni.py (hunk 4 above)
Comment (Guberlo) [reply module/lezioni.py:97]: Actually, yes! 
It's like the key 'sessioneprima sessione' in esami.

Going to fix this as soon as possible

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
What about merge? @TendTo

**Comment by Guberlo** [comment]
@Helias Actually, I still have to fix some minor things (right now I need to remove some unused imports in callback_handlers.py and move help text to MD file)

I just need some more minutes (atm I'm having dinner)

**Comment by Guberlo** [comment]
I should have fixed everything now, also updated help.md with some missing emojis /functions

**Review by TendTo — APPROVED** [review]

**Pull request merged**